### PR TITLE
Fix CI.yml CI/CD

### PR DESCRIPTION
**Pull request by ellioty**
This PR is meant to fix issues that were not properly addressed in the previous PR: https://github.com/FuelLabs/fuel-core/pull/271

Essentially the Docker step has a dependency on Publishing to Crates, however, we manually ignore this since the Docker step has `if: always()` so all dependencies within the Docker step are currently being ignored.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -162,7 +162,7 @@ jobs:
       - cancel-previous-runs
       - lint-toml-files
       - cargo-verifications
-      - publish-crates
+    # - publish-crates
```
Comment (adlerjohn) [inline .github/workflows/ci.yml:165]: Why is this commented out?

Comment (Voxelot) [reply .github/workflows/ci.yml:165]: Agreed, we should not be disabling our entire publishing pipeline from master for the sake of debugging this deployment.

**Comment by adlerjohn** [comment]
Can you update the PR description @ellioty with some information of what this PR is actually for?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ellioty) [reply .github/workflows/ci.yml:165]: This is only a dependency, which actually isn't even being respected, if you see 2 lines below we're doing `if: always()`

I actually meant to completely remove it

**Review by Voxelot — APPROVED** [review]

**Pull request merged**